src/network/netdev/tunnel.c

```diff
@@ -600,6 +640,41 @@ int config_parse_encap_limit(const char* unit,
         return 0;
 }
 
+int config_parse_tunnel_fou_port(const char *unit,
+                                 const char *filename,
+                                 unsigned line,
+                                 const char *section,
+                                 unsigned section_line,
+                                 const char *lvalue,
+                                 int ltype,
+                                 const char *rvalue,
+                                 void *data,
+                                 void *userdata) {
+        Tunnel *t = userdata;
+        uint16_t port;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (!STR_IN_SET(rvalue, "0", "auto")) {
```

Comment:
I don't think there's any need to accept "0" in addition to "auto". Let's only permit "auto".